index.html

```diff
@@ -0,0 +1,464 @@
+            }
+          }
+        </pre>
+      </section>
+    </section>
+    <section>
+      <h2>
+        Handling Web App Launches
+      </h2>
+      <section>
+        <h2>
+          Launching a Web App
+        </h2>
+        <p>
+          When launching a web app via any means (e.g. OS launch, protocol handling, file handling, share target, link capturing, etc.) the trigger of the launch should run the steps to [=launch a web app=] with a {{LaunchParams}} containing relevant launch information (e.g. share data in the event of a share target launch).
+        </p>
+        <p>
+          The steps to <dfn>launch a web app</dfn> are given by the following algorithm and takes a {{LaunchParams}} |params:LaunchParams|.
+        </p>
+        <ol class="algorithm">
+          <li>If |params| is null, set |params| to a new {{LaunchParams}} with {{LaunchParams/targetURL}} set to [=manifest/start_url=].
+          </li>
+          <li>Set |client| to the result of running the steps to [=prepare a web app launch client=] passing [=manifest/route_to=] and |params|.{{LaunchParams/targetURL}}.
+          </li>
+          <li>Append |params| to the [=unconsumed launch params=] of the |client|'s document's {{Window.LaunchQueue}}.
+          </li>
+          <li>If the [=assigned launch consumer=] |consumer| is set:
+            <ol>
+              <li>[=list/For each=] |launch_params:LaunchParams| of [=unconsumed launch params=]:
+                <ol>
+                  <li>Invoke |consumer| with |launch_params|.
+                  </li>
+                </ol>
+              </li>
+              <li>Set [=unconsumed launch params=] to the empty list.
+              </li>
+            </ol>
+          </li>
+        </ol>
+      </section>
+      <section>
+        <h2>
+          Choosing a [=Web App Launch Client=]
+        </h2>
+        <p>
+          A <dfn>web app launch client</dfn> is a [=service worker client/window client=] associated with the web app.
+        </p>
+        <p class="note">
+          The exact form of this association is up to the user agent e.g. a dedicated app window separated in its UI presentation from general hyperlink browsing.
+        </p>
+        <p>
+          To <dfn>prepare a web app launch client</dfn>, given |route to| and |target URL:URL|:
+        </p>
+        <ol class="algorithm">
+          <li>
+            <p>Switching on |route to|, run the following substeps:</p>
+            <dl class="switch">
+              <dt>[=route to/auto=]</code>
+              <dd>
+                <ol class="algorithm">
+                  <li>Let |route_to| be either [=route to/new-client=] or [=route to/existing-client-navigate=] according to the user agent's decision for which is most appropriate.
+                  </li>
+                  <li>Return the result of the steps to [=prepare a web app launch client=] passing |route to| and |target URL|.
```

Comment:
The recursive call is a little tricksy. Could we instead just put a "Step 1: If _route to_ is auto, let _route to_ be either ...", then "Step 2" can be to switch on _route to_", and not handle the "auto" case?

Another idea is to actually perform this auto default selection process in the "steps to process the route_to member", putting the chosen default into the processed manifest there, so a processed manifest will never have "auto".

Reply:
Moved auto handling out of switch statement.